mibidata/mibi_image.py

```diff
@@ -91,57 +115,66 @@ class MibiImage():
         mass_gain: Mass gain used for mass calibration.
         time_resolution: Parameter used for mass calibration.
-        miscalibrated: Whether or not there was significant difference between
-            peak locations after mass recalibration.
+        miscalibrated: Whether or not there was significant difference
+            between peak locations after mass recalibration.
         check_reg: Whether or not the maximum shift between depths is higher
             than a threshold.
-        filename: The name of the Run XML file which corresponds to the run
-            name.
+        filename: The name of the file with the run metadata.
+        description: String describing the image.
     """
 
-    # pylint: disable=too-many-arguments
-    def __init__(self, data, channels, run=None, date=None,
-                 coordinates=None, size=None, slide=None, point_name=None,
-                 folder=None, dwell=None, scans=None, aperture=None,
-                 instrument=None, tissue=None, panel=None, version=None,
-                 mass_offset=None, mass_gain=None, time_resolution=None,
-                 miscalibrated=None, check_reg=None,
-                 datetime_format=_DATETIME_FORMAT, filename=None):
+    def __init__(self, data, channels, **kwargs):
 
+        # initialize non metadata attributes
         self._length = len(channels)
         if data.shape[2] != self._length:
             raise ValueError('Channels length does not match data dimensions.')
         self.data = data
         self._set_channels(channels, self._length)
 
+        # initialize expected metadata
+        # if no version specified, assume current version
+        self.version = kwargs.pop('version', MIBITIFF_VERSION)
+        date = kwargs.pop('date', None)
+        datetime_format = kwargs.pop('datetime_format', _DATETIME_FORMAT)
         try:
             self.date = datetime.datetime.strptime(date, datetime_format)
         except TypeError:  # Given as datetime obj already, or None.
             self.date = date
-
-        self.run = run
-        self.coordinates = coordinates
-        self.size = size
-        self.slide = slide
-        self.point_name = point_name
-        self.folder = folder
-        self.dwell = dwell
-        self.scans = scans
-        self.aperture = aperture
-        self.instrument = instrument
-        self.tissue = tissue
-        self.panel = panel
-        self.version = version
-        self.mass_offset = mass_offset
-        self.mass_gain = mass_gain
-        self.time_resolution = time_resolution
-        self.miscalibrated = miscalibrated
-        self.check_reg = check_reg
-        self.filename = filename
+        self.run = kwargs.pop('run', None)
+        self.coordinates = kwargs.pop('coordinates', None)
+        self.size = kwargs.pop('size', None)
+        self.slide = kwargs.pop('slide', None)
+        self.fov_id = kwargs.pop('fov_id', None)
+        self.fov_name = kwargs.pop('fov_name', None)
```

Comment:
One of the subtasks of the ticket was: "MibiImge constructor for backwards compatibility to change point_name keyword to fov_name" which I don't see implemented here.

Reply:
I check for the old metadata format and convert to the new format. It's done in `_convert_from_previous_metadata_versions`.